marlowe-playground-client/src/Marlowe/Monaco.purs

```diff
@@ -11,10 +11,14 @@ module Marlowe.Monaco
 import Prelude
 import Data.Function.Uncurried (Fn1, runFn1)
 import Data.Maybe (Maybe(..))
+import Data.Unfoldable as Unfoldable
 import Halogen (RefLabel(..))
 import Halogen.Monaco (Settings)
+import Help as Help
 import Marlowe.Linter as Linter
-import Monaco (CodeAction, CodeActionProvider, CompletionItem, CompletionItemProvider, DocumentFormattingEditProvider, Editor, IMarkerData, IRange, IStandaloneThemeData, LanguageExtensionPoint(..), Theme, TokensProvider, Uri)
+import Monaco (CodeAction, CodeActionProvider, CompletionItem, CompletionItemProvider, DocumentFormattingEditProvider, Editor, HoverProvider, IMarkdownString, IMarkerData, IRange, IStandaloneThemeData, LanguageExtensionPoint(..), Theme, TokensProvider, Uri)
+
+foreign import hoverProvider_ :: Fn1 (String -> { contents :: Array IMarkdownString }) HoverProvider
```

Comment:
What's the point of `Fn1` btw? Is it better than just `(String -> { contents :: Array IMarkdownString }) -> HoverProvider`?

Reply:
I had a look at changing it and it doesn't really make much difference, the reason it occurred in the first place is because some of the other `Fn1`s were previously `FnN`s that got changed. This leaves a consistent pattern for using foreign functions no matter how many arguments they have.